client/landing/stepper/declarative-flow/internals/steps-repository/import-with-site-address/index.tsx

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+import { isEnabled } from '@automattic/calypso-config';
+import React from 'react';
+import CaptureStep from 'calypso/blocks/import/capture';
+import CaptureStepRetired from 'calypso/blocks/import/capture-retired';
+import { ImportWrapper } from '../import';
+import { generateStepPath } from '../import/helper';
+import type { Step } from '../../types';
+
+import './style.scss';
+
+const isEnabledImportLight = isEnabled( 'onboarding/import-light-url-screen' );
+
+const ImportWithSiteAddressStep: Step = function ImportStep( props ) {
+	const { navigation } = props;
+
+	return (
+		<ImportWrapper { ...props }>
+			{ isEnabledImportLight ? (
+				<CaptureStep
+					disableImportListStep
+					goToStep={ ( step, section ) =>
+						navigation.goToStep?.( generateStepPath( step, section ) )
+					}
+				/>
+			) : (
+				<CaptureStepRetired
+					goToStep={ ( step, section ) =>
+						navigation.goToStep?.( generateStepPath( step, section ) )
+					}
+				/>
+			) }
+		</ImportWrapper>
+	);
+};
+
+export default ImportWithSiteAddressStep;
```

Comment:
Since here we introduced a new Step whose purpose is entering the source site address; I'm wondering what the difference is between this step and the step we already have here: https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx

It seems like exactly the same steps. Isn't it?

Reply:
The extra component was added to pass in `disableImportListStep` as we're not supporting the `ImportList` step in this iteration. 

I couldn't see a way to override this. Ideally this step would have taken `props`:

```ts
	useSteps() {
		return [
			{ slug: 'import', component: ImportWithSiteAddressStep }, //pass props here with component name
			
			```

Reply:
I think that we should avoid making another import/capture step doing the same stuff.

In the regular `import` step [where we set up `CaptureStep`](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx#L67) you can provide additional props based on the flow that consumes the step at the moment.

```
const { navigation, flow } = props;

<CaptureStep
	disableImportListStep={ IMPORT_HOSTED_SITE_FLOW === flow } // <------- solution
	goToStep={ ( step, section ) =>
		navigation.goToStep?.( generateStepPath( step, section ) )
	}
/>
```

@mpkelly what do you think?

Reply:
Ok, let's go with that.